debbuild

```diff
@@ -172,6 +193,10 @@ if ($cmdopts{type} eq 't') {
 }
 
 if ($cmdopts{type} eq 'b') {
+  # Make a copy of spec lowercased and `_` replaced with `-` to do some magic
+  (my $spec_lc = lc(basename($specglobals{specfile}))) =~ tr/_/-/;
+  $spec_lc = dirname($specglobals{specfile})."/".$spec_lc;
+  qx ( $specglobals{__cp} -a $specglobals{specfile} $spec_lc ) if ( $specglobals{specfile} ne $spec_lc );
```

Comment:
Why are we doing this?

Reply:
Most of python modules spec files are processed by Lua macro and it tries open the spec file directly from fs. In case if the name of the spec file is not matched the package name (for example if the package name contains uppercased letters or `-`) it will fail to open the spec file.

Reply:
So this is a bug in your Lua macros then?

Reply:
No, it's not a bug in Lua macros. The package name and `spec` could be named with uppercase in RPM. For example quite common to have `python-SomeModule`, but dpkg is building all the packages lowercased, but if we put the package names in the dependencies with uppercased letters it will fail to find out the package. In this case we need to lowercase and translate all the `-` to `_` and it's better to do it initially before processing all the macros just to prevent putting uppercased names in any place. But Lua macros for python mostly used to parse the spec itself and usually open it by the value from `_specfile` macro or by the name of the package (`python-SomeModule.spec` for example), but on modifying the package name we will get the value like `python_somemodule.spec` and this file is not there as we generated the new name.

Reply:
But debbuild itself should not care about this. I have perl module packages that are "normal" for rpm that produce perfectly sensible SDEBs and build reasonable DEBs themselves.

Reply:
perl module packages are not using lua macroses trying to open spec files directly, but some of the python modules in OBS is using this way